webapp/src/immersivereader.tsx

```diff
@@ -46,29 +62,98 @@ function beautifyText(content: string): string {
         );
     }
 
-    // Remove ** around bold text, replace with "
-    function cleanBold(content: string): string {
+    // If the ```block ``` annotation is used to display a block in
+    // a modal, remove it
+    function cleanHintText(content: string): string {
         return content.replace(
-            /\*\*([^*]+)\*\*/gu,
-            (matched, word, offset, s) => lf("\"{0}\"", word)
+            /^```block$[^`]*^```$/gum,
+            ""
+        );
+    }
+
+    // Remove ** and __ around bold text, replace with " if it's only one word
+    function cleanBold(content: string): string {
+
+        let singleWordBold = content.replace(
+            /\*\*([^*^\s]+)\*\*|__([^\s^_]+)__/gu,
```

Comment:
nit: feel free to ignore -- if you want to avoid the ternary / matching two groups you could use this regex instead here:

``/(?:\*\*|__)([^*^\s]+)(?:\*\*|__)/``

?: at the start of a group denotes it as a 'non capture group' / gets skipped for the groupings. Applies to both this fns regexes and the clean italics I think?

Reply:
oooh yeah I didn't do this because then it could grab a **(wordswordswords)__ case, I think?

Reply:
but also I didn't konw about ?: lol

Reply:
Ohhh good point, then maybe use a backref `/(\*\*|__)([^*^\s]+)\1/gu`? have to skip the first in the replace function but don't have to do ternaries in each replace later on:

https://regex101.com/r/iwUPLA/1

Reply:
:O that's amazingggg